normandy/control/static/control/js/components/RecipePreview.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import composeRecipeContainer from './RecipeContainer.jsx'
+import {runRecipe} from '../../../../../selfrepair/static/js/self_repair_runner.js';
+
+class RecipePreview extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      recipeExecuted: false,
+    };
+  }
+
+  componentWillMount() {
+    this.attemptPreview();
+  }
+
+  componentDidUpdate() {
+    this.attemptPreview();
+  }
+
+  attemptPreview() {
+    const {recipe} = this.props;
+    const {recipeExecuted} = this.state;
+
+    if (recipe && !recipeExecuted) {
+      runRecipe(recipe, {testing: true}).catch(err => {
+        console.error(err);
+      });
+
+      this.setState({recipeExecuted: true});
+    }
+  }
+
+  render() {
+    const {recipe} = this.props;
+    if (recipe) {
+      return (
+        <div className="fluid-7">
```

Comment:
One thing I didn't get to with my branch was figuring out if there's something more useful we can put here than just a preview message. Maybe a fake console? A message when the recipe is finished executing? I dunno.